abstract class Shape{
    abstract double area();
    void displayArea(){
        System.out.printf("Area = %.2f\n",area());
    }
}
class Circle extends Shape{
    double radius;

    Circle(double radius){
        this.radius = radius;
    }
    double area(){
        return Math.PI * radius *radius;
    }
}
class Rectangle extends Shape{
    double length;
    double width;

    Rectangle(double length, double width){
        this.length =length;
        this.width = width;
    }
    double area(){
        return length*width;
    }
}

public class UseCase2 {
    public static void main(String[] args) {
        Shape[] shapes = new Shape[2];
        shapes[0] = new Circle(5.0);
        shapes[1] = new Rectangle(4.0, 6.0);
 
        for (Shape s : shapes) {
            s.displayArea();   // must print correct area for each shape
        }
    }

}
